network-config-analyzer/NetworkConfigQuery.py

```diff
@@ -826,26 +824,35 @@ class ContainmentQuery(TwoNetworkConfigsQuery):
 
-        peers_to_compare = self.config1.peer_container.get_all_peers_group()
-        peers_to_compare |= self.disjoint_referenced_ip_blocks()
-        captured_pods = self.config1.get_captured_pods() | self.config2.get_captured_pods()
-        for peer1 in peers_to_compare:
-            for peer2 in peers_to_compare if peer1 in captured_pods else captured_pods:
-                if peer1 == peer2:
-                    continue
-                captured1, conns1, _ = self.config1.allowed_connections(peer1, peer2, only_captured)
-                if only_captured and not captured1:
-                    continue
-                _, conns2, _ = self.config2.allowed_connections(peer1, peer2)
-                if not conns1.contained_in(conns2):
-                    output_result = f'{self.name1} is not contained in {self.name2}'
-                    output_explanation = f'Allowed connections from {peer1} to {peer2} in {self.name1} ' \
-                                         f'are not a subset of those in {self.name2}\n'
-                    output_explanation += conns1.print_diff(conns2, self.name1, self.name2)
-                    return QueryAnswer(False, output_result, output_explanation)
+        config1_peers = self.config1.peer_container.get_all_peers_group()
+        captured1_pods = self.config1.get_captured_pods()
+
+        all_ip_blocks = PeerSet()
+        all_ip_blocks.add(IpBlock.get_all_ips_block())
+        config1_ip_blocks = self.disjoint_ip_blocks(self.config1.get_referenced_ip_blocks(), all_ip_blocks)
+        config2_ip_blocks = self.disjoint_ip_blocks(self.config2.get_referenced_ip_blocks(), all_ip_blocks)
+        disjoint_ip_blocks = self.disjoint_ip_blocks(config1_ip_blocks, config2_ip_blocks)
+
+        peers_to_compare = config1_peers | disjoint_ip_blocks
+        for pair in itertools.permutations(peers_to_compare, 2):
+            if only_captured and pair[0] not in captured1_pods and pair[1] not in captured1_pods:
+                continue
+            captured1, conns1, _ = self.config1.allowed_connections(pair[0], pair[1], only_captured)
+            # TODO: understand why next 'if only_captured and not captured1' is required, since anyhow we check
+            #  that one of the peers is in captured1_pods
+            #  without it we fail on some calico tests
+            if only_captured and not captured1:
+                continue
+            _, conns2, _ = self.config2.allowed_connections(pair[0], pair[1])
+            if not conns1.contained_in(conns2):
+                output_result = f'{self.name1} is not contained in {self.name2}'
+                output_explanation = f'Allowed connections from {pair[0]} to {pair[1]} in {self.name1} ' \
+                                     f'are not a subset of those in {self.name2}\n'
+                output_explanation += conns1.print_diff(conns2, self.name1, self.name2)
+                return QueryAnswer(False, output_result, output_explanation)
```

Comment:
We have to discuss this. I thought we can just check if `config1.peer_container` is contained in `config.peer_container` and then continue as before. Why isn't it the case?

Reply:
done per discussion